chain/src/pipe.rs

```diff
@@ -245,9 +245,16 @@ fn validate_header(header: &BlockHeader, ctx: &mut BlockContext) -> Result<(), E
 		return Err(Error::InvalidBlockTime);
 	}
 
+	// TODO - confirm genesis height is 1
+	let prev_header = if header.height > 1 {
+		Some(ctx.store.get_block_header(&header.previous)?)
+	} else {
+		None
+	};
+
 	if !ctx.opts.contains(Options::SKIP_POW) {
 		let n = global::sizeshift() as u32;
-		if !(ctx.pow_verifier)(header, n) {
+		if !(ctx.pow_verifier)(header, prev_header, n) {
```

Comment:
I think it's better to have PoW verification not depend on any previous block and be self-sufficient. Then pull up the actual `pow > difficulty` test here. It also means you can validate the PoW before loading the previous block above.

Reply:
I think I follow what you're saying. Let me rework this and get back to you.